src/Apps/Order/Routes/Shipping2/AddressModal2.jest.tsx

```diff
@@ -0,0 +1,340 @@
+import { setupTestWrapper } from "DevTools/setupTestWrapper"
+import { graphql } from "react-relay"
+import { AddressModal2TestQuery } from "__generated__/AddressModal2TestQuery.graphql"
+
+jest.unmock("react-relay")
+jest.mock("System/useSystemContext")
+jest.mock("Utils/Hooks/useMatchMedia", () => ({
+  __internal__useMatchMedia: () => ({}),
+}))
+jest.mock("Utils/user", () => ({
+  userHasLabFeature: jest.fn(),
+}))
+
+jest.mock("Apps/Order/Routes/Shipping2/Hooks/useParseOrderData", () => {
+  return {
+    useParseOrderData: () => ({
+      shipsFrom: "US",
+    }),
+  }
+})
+
+const errorBoxQuery = "Banner[data-testid='form-banner-error']"
+
+const savedAddress: SavedAddressType = {
+  ...validAddress,
+  phoneNumber: "8475937743",
+  id: "id",
+  internalID: "internal-id",
+  addressLine3: null,
+  isDefault: false,
+}
+
+let testAddressModalProps: Props
+
+let mockRelayEnv: ReturnType<typeof createMockEnvironment>
+
+beforeEach(() => {
+  mockRelayEnv = createMockEnvironment()
+  ;(useSystemContext as jest.Mock).mockImplementation(() => {
+    return {
+      user: { lab_features: [] },
+      isLoggedIn: true,
+      relayEnvironment: mockRelayEnv,
+    }
+  })
+})
+
+const { getWrapper } = setupTestWrapper<AddressModal2TestQuery>({
+  // Cast props here because we don't actually need the query, just the relay
+  // test wrapper. -
+  // props must be passed via getWrapper's componentProps arg
+  Component: (props: unknown) => {
+    return (
+      <ShippingContext.Provider value={useComputeShippingContext({} as any)}>
+        <AddressModal {...(props as Props)} />
+      </ShippingContext.Provider>
+    )
+  },
+  query: graphql`
+    query AddressModal2TestQuery @relay_test_operation {
+      _unused: artist(id: "whocare") {
+        name
+      }
+    }
+  `,
+})
+
+describe("AddressModal", () => {
+  beforeEach(() => {
+    testAddressModalProps = {
+      onSuccess: jest.fn(),
+      modalAction: {
+        type: AddressModalActionType.EDIT_USER_ADDRESS,
+        address: savedAddress,
+      },
+
+      closeModal: jest.fn(),
+    }
+    ;(useSystemContext as jest.Mock).mockImplementation(() => {
+      return {
+        user: { lab_features: [] },
+        isLoggedIn: true,
+        relayEnvironment: mockRelayEnv,
+      }
+    })
+  })
+
+  it("renders EditModal with the title, input fields and buttons", () => {
+    const { wrapper } = getWrapper({}, testAddressModalProps, mockRelayEnv)
+
+    expect(wrapper.text()).toContain("Edit address")
+    expect(wrapper.find("input").length).toBe(7)
+    expect(wrapper.find("select").length).toBe(1)
+
+    expect(wrapper.find("Checkbox[data-test='setAsDefault']").length).toBe(1)
+    expect(wrapper.find("Clickable[data-test='deleteButton']").length).toBe(1)
+    expect(wrapper.find("Button[data-test='saveButton']").length).toBe(1)
+  })
+
+  it("renders EditModal without checkbox when address is default", () => {
+    const { wrapper } = getWrapper(
+      {},
```

Comment:
this empty object first arg is super weird, lets make `getWrapper` take a single object and then just pluck what we need from that:

```tsx
const getWrapper = (props) => {
  // props.foo, props.bar 
}
```

Reply:
Updated to redeclare the function locally, lmk if that's what you have in mind (once I push)